File: NInja C, CPP, SDL2/if you want to edit game/intro.cpp
#include "intro.h"
#include "font.h"
#include <stdio.h>

Intro::Intro()
{
    counter = 0;
    quit = false;

    x = NULL;
    y = NULL;
    texture = NULL;
}

Intro::~Intro()
{
    free();
}


void Intro::free()
{
    counter = 0;
    quit = false;

    if( texture != NULL )
    {
        for( int i = 0; i < nr; i++ )
        {
            texture[ i ].free();
        }

        delete [] texture;
        texture = NULL;
    }

    if( x != NULL )
    {
        delete [] x;
        x = NULL;
    }

    if( y != NULL )
    {
        delete [] y;
        y = NULL;
    }
}

bool Intro::isClose()
{
    return quit;
}


bool Intro::load( SDL_Renderer* &renderer, SDL_Window* &window )
{
    bool success = true;

    free();

    int w, h;
    SDL_GetWindowSize( window, &w, &h );

    Font* font = new Font;
    if( !font->load( "intro/MaszynaAEG.ttf", 37 ) )
    {
        success = false;
    }
    else
    {
        x = new int [ nr ];
        if( x == NULL )
        {
            printf( "Not created array of int\n" );
            success = false;
        }

        y = new int [ nr ];
        if( y == NULL )
        {
            printf( "Not created array of int\n" );
            success = false;
        }

        texture = new Texture[ nr ];
        if( texture == NULL )
        {
            printf( "Not created array of texture\n" );
            success = false;
        }


        // Our background texture
        if( !texture[ BACKGROUND ].createWithColor( renderer, 0x15, 0x15, 0x1D, w, h ) )
        {
            success = false;
        }
        else
        {
            x[ BACKGROUND ] = 0;
            y[ BACKGROUND ] = 0;
        }


        // Our text on start
        SDL_Color white = { 0xFF, 0xFF, 0xFF };
        if( !texture[ PREVIEW ].loadFromRenderedText( renderer, font->get(), "When nothing makes sense...", white ) )
        {
            success = false;
        }
        else
        {
            x[ PREVIEW ] = (w / 2) - (texture[ PREVIEW ].getWidth() / 2);
            y[ PREVIEW ] = (h / 2) - (texture[ PREVIEW ].getHeight() / 2);
        }


        // Title and image
        SDL_Color green = { 0x70, 0xB7, 0x59 };
        if( !texture[ TITLE ].loadFromRenderedText( renderer, font->get(), "Jump and jump", green ) )
        {
            success = false;
        }
        else if( !texture[ SHURIKEN ].loadFromFile( renderer, window, "intro/shuriken.png" ) )
        {
            success = false;
        }
        else
        {
            x[ TITLE ] =    (w / 2) - (texture[ TITLE ].getWidth() / 2) - 10 - texture[ SHURIKEN ].getWidth();
            y[ TITLE ] =    (h / 2) - (texture[ TITLE ].getHeight() / 2);
            x[ SHURIKEN ] = x[ TITLE ] + texture[ TITLE ].getWidth() + 10;
            y[ SHURIKEN ] = (h / 2) - (texture[ SHURIKEN ].getHeight() / 2);
        }


        // My name and text
        if( !texture[ AUTHOR ].loadFromRenderedText( renderer, font->get(), "Adrian Micha³ek", white ) )
        {
            success = false;
        }
        else if( !texture[ PRODUCED ].loadFromRenderedText( renderer, font->get(), "produced by", green ) )
        {
            success = false;
        }
        else
        {
            x[ AUTHOR ] =   (w / 2) - (texture[ AUTHOR ].getWidth() / 2);
            y[ AUTHOR ] =   (h / 2) - (texture[ AUTHOR ].getHeight() / 2);
            x[ PRODUCED ] = (w / 2) - (texture[ PRODUCED ].getWidth() / 2);
            y[ PRODUCED ] = (h / 2) - (texture[ PRODUCED ].getHeight() / 2) - texture[ AUTHOR ].getHeight() - 10;
        }
    }

    delete font;
    font = NULL;

    return success;
}

void Intro::render( SDL_Renderer* &renderer, int fps )
{
    static float seconds = 1.5;
    static int frames = 3;

    counter ++;

    static Uint8 alpha = 0;

    if( alpha < 250 )
        alpha ++;

    for( int i = 1; i <= frames; i++ )
    {
        if( counter == fps * ( seconds * ( frames - ( frames - i ) ) ) )
        {
            alpha = 0;
        }
    }

    for( int i = BACKGROUND; i < nr; i++ )
    {
        texture[ i ].setAlpha( alpha );
    }

    texture[ BACKGROUND ].render( renderer, x[ BACKGROUND ], y[ BACKGROUND ] );

    if( counter < fps * ( seconds * ( frames - 2 ) ) )
    {
        texture[ PREVIEW ].render( renderer, x[ PREVIEW ], y[ PREVIEW ] );
    }

    else if( counter < fps * ( seconds * ( frames - 1 ) ) )
    {
        texture[ SHURIKEN ].render( renderer, x[ SHURIKEN ], y[ SHURIKEN ] );
        texture[ TITLE ].render( renderer, x[ TITLE ], y[ TITLE ] );
    }

    else if( counter < fps * ( seconds * frames ) )
    {
        texture[ AUTHOR ].render( renderer, x[ AUTHOR ], y[ AUTHOR ] );
        texture[ PRODUCED ].render( renderer, x[ PRODUCED ], y[ PRODUCED ] );
    }

    else
    {
        quit = true;
    }
}

File: intro.py
#!/usr/bin/python

import pygame
from texture import Texture
from text import Text

class Intro:

	'''
	nie jest to wersja finalowa aplikacji
	gra moze byc rozwijana rowniez dzieki tobie
	kod dziala na azsadzie open source
	aplikacja prezentuje wykorzystanie biblioteki pygame
	'''

#-------------------------------------------------------------------------------------------------------
	
	def __init__( self ):

		self.counter = 0
		self.quit = False

		#Textures
		self.preview = Text( "intro/Jaapokki-Regular.otf", 37 )
		self.shuriken = Texture( "intro/shuriken.png" )
		self.title = Text( "intro/Jaapokki-Regular.otf", 37 )
		self.author = Text( "intro/Jaapokki-Regular.otf", 37 )
		self.produced = Text( "intro/Jaapokki-Regular.otf", 37 )

		self.text_one = Text( "intro/Jaapokki-Regular.otf", 37 )
		self.text_two = Text( "intro/Jaapokki-Regular.otf", 37 )
		self.text_three = Text( "intro/Jaapokki-Regular.otf", 37 )
		self.text_four = Text( "intro/Jaapokki-Regular.otf", 37 )

#-------------------------------------------------------------------------------------------------------
	
	def load( self, width, height ):

		self.preview.createText( "When nothing makes sense...", [ 0xFF, 0xFF, 0xFF ] )
		self.preview.setX( width/2 - self.preview.getWidth()/2 )
		self.preview.setY( height/2 - self.preview.getHeight()/2 )

		self.title.createText( "Ninja", [ 0x70, 0xB7, 0x59 ] )
		self.title.setX( width/2 - self.title.getWidth()/2 - self.shuriken.getWidth() -10 )
		self.title.setY( height/2 - self.title.getHeight()/2 )
		
		
		self.shuriken.setX( self.title.getRight() +10 )
		self.shuriken.setY( height/2 - self.shuriken.getHeight()/2 )
		
		self.author.createText( "Adrian Michalek", [ 0xFF, 0xFF, 0xFF ] )
		self.author.setX( width/2 - self.author.getWidth()/2 )
		self.author.setY( height/2 - self.author.getHeight()/2 )

		self.produced.createText( "produced by", [ 0x70, 0xB7, 0x59 ] )
		self.produced.setX( width/2 - self.produced.getWidth()/2 )
		self.produced.setY( height/2 - self.produced.getHeight()/2 - self.author.getHeight() -10 )

		
		self.text_one.createText( "It's not a final version.", [ 0xFF, 0xFF, 0xFF ] )
		self.text_one.setX( width/2 - self.text_one.getWidth()/2 )
		self.text_one.setY( height/2 - self.text_one.getHeight()-20 )

		self.text_two.createText( "You can develop game by your own.", [ 0xFF, 0xFF, 0xFF ] )
		self.text_two.setX( width/2 - self.text_two.getWidth()/2 )
		self.text_two.setY( height/2 - self.text_two.getHeight()/2-10 )

		self.text_three.createText( "It's a open source code build with pygame library.", [ 0xFF, 0xFF, 0xFF ] )
		self.text_three.setX( width/2 - self.text_three.getWidth()/2 )
		self.text_three.setY( height/2 )

		self.text_four.createText( "Have a nice time!", [ 0x70, 0xB7, 0x59 ] )
		self.text_four.setX( width/2 - self.text_four.getWidth()/2 )
		self.text_four.setY( height/2 + self.text_four.getHeight()/2 +10 )

#-------------------------------------------------------------------------------------------------------

	def free( self ):
		#DELETE ALL
		del self.counter
		del self.quit
		
		self.preview.free()
		del self.preview

		self.shuriken.free()
		del self.shuriken
		
		self.title.free()
		del self.title
		
		self.author.free()
		del self.author

		self.produced.free()
		del self.produced
		
		self.text_one.free()
		del self.text_one

		self.text_two.free()
		del self.text_two

		self.text_three.free()
		del self.text_three

		self.text_four.free()
		del self.text_four

#-------------------------------------------------------------------------------------------------------
	
	def draw( self, screen ):
		
		self.preview.fadein( 3 )
		if self.preview.getAlpha() < 255:
			self.preview.draw( screen )
		else:
			self.title.fadein( 3 )
			self.shuriken.fadein( 3 )
	
		if self.title.getAlpha() < 255:
			self.title.draw( screen )
			self.shuriken.draw( screen )
		else:
			self.author.fadein( 3 )
			self.produced.fadein( 3 )

		if self.author.getAlpha() < 255:
			self.author.draw( screen )
			self.produced.draw( screen )
		else:
			self.counter += 1
			self.text_one.fadein( 3 )
			self.text_two.fadein( 3 )
			self.text_three.fadein( 3 )
			self.text_four.fadein( 3 )
		if self.counter < 500:
			self.text_one.draw( screen )
			self.text_two.draw( screen )
			self.text_three.draw( screen )
			self.text_four.draw( screen )
		else:
			self.quit = True

#-------------------------------------------------------------------------------------------------------

	def getQuit( self ):
		return self.quit

#-------------------------------------------------------------------------------------------------------
